src/types.ts

```diff
@@ -0,0 +1,335 @@
+export type Nullable<T> = T | null;
+
+export type CallbackComponentReturnValue<T = void> = T | Promise<T>;
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
```

Comment:
Maybe we should disable this rule globally

Reply:
For now I just them there. Let's see how it goes and then eventually we revisit this.